workflows/detaxizer.nf

```diff
@@ -1,129 +1,470 @@
+    //
+    // MODULE: Run fastp
+    //
+    FASTP (
+        RENAME_FASTQ_HEADERS_PRE.out.fastq,
+        [],
+        params.fastp_save_trimmed_fail,
+        []
     )
+    ch_versions = ch_versions.mix(FASTP.out.versions.first())
 
     //
-    // MODULE: MultiQC
+    // MODULE: Prepare Kraken2 Database
     //
-    workflow_summary    = WorkflowDetaxizer.paramsSummaryMultiqc(workflow, summary_params)
-    ch_workflow_summary = Channel.value(workflow_summary)
+    ch_kraken2_db = Channel.fromPath(params.kraken2db).map {
+            item -> [['id': "kraken2_db"], item]
+        }
+    KRAKEN2PREPARATION (
+        ch_kraken2_db
+    )
+    ch_versions = ch_versions.mix(KRAKEN2PREPARATION.out.versions)
 
-    methods_description    = WorkflowDetaxizer.methodsDescriptionText(workflow, ch_multiqc_custom_methods_description, params)
-    ch_methods_description = Channel.value(methods_description)
+    //
+    // MODULE: Run Kraken2
+    //
 
-    ch_multiqc_files = Channel.empty()
-    ch_multiqc_files = ch_multiqc_files.mix(ch_workflow_summary.collectFile(name: 'workflow_summary_mqc.yaml'))
-    ch_multiqc_files = ch_multiqc_files.mix(ch_methods_description.collectFile(name: 'methods_description_mqc.yaml'))
-    ch_multiqc_files = ch_multiqc_files.mix(CUSTOM_DUMPSOFTWAREVERSIONS.out.mqc_yml.collect())
-    ch_multiqc_files = ch_multiqc_files.mix(FASTQC.out.zip.collect{it[1]}.ifEmpty([]))
+    KRAKEN2_KRAKEN2 (
+        FASTP.out.reads,
+        KRAKEN2PREPARATION.out.db.first(),
+        params.save_output_fastqs,
+        true
+    )
+    ch_versions = ch_versions.mix(KRAKEN2_KRAKEN2.out.versions.first())
+
+    //
+    // MODULE: Parse the taxonomy from the kraken2 report and return all subclasses of the tax2filter
+    //
+    PARSE_KRAKEN2REPORT(
+        KRAKEN2_KRAKEN2.out.report.take(1)
+    )
+    ch_versions = ch_versions.mix(PARSE_KRAKEN2REPORT.out.versions)
+
+    //
+    // MODULE: Isolate the hits for a certain taxa and subclasses
+    //
+    ch_parsed_kraken2_report = PARSE_KRAKEN2REPORT.out.to_filter.map {meta, path -> path}
+
+    KRAKEN2_KRAKEN2.out.classified_reads_assignment.combine(ch_parsed_kraken2_report).set{ ch_combined }
+
+    ISOLATE_IDS_FROM_KRAKEN2_TO_BLASTN (
+        ch_combined
+    )
+
+    ch_versions = ch_versions.mix(ISOLATE_IDS_FROM_KRAKEN2_TO_BLASTN.out.versions.first())
+
+    //
+    // MODULE: Summarize the kraken2 results and the isolated kraken2 hits
+    //
+    ch_prepare_summary_kraken2 = KRAKEN2_KRAKEN2.out.classified_reads_assignment.join(ISOLATE_IDS_FROM_KRAKEN2_TO_BLASTN.out.classified).map {
+        meta, path1, path2 ->
+            def newMeta = meta.clone()
+            return [ newMeta, [ path1, path2 ] ]
+    }
+
+    ch_combined_kraken2 = ch_prepare_summary_kraken2.map {
+        meta, path ->
+            def newMeta = meta.clone()
+            newMeta.id = newMeta.id.replaceAll("(_R1|_R2)", "")
+            return [ newMeta , path]
+        }
+        .groupTuple(by: [0])
```

Comment:
adding the grouping size would be great. otherwise you are creating a bottleneck here. If  the groups have variable sizes you can also use `groupKey`